Aquarius/Src/Aquarius/Events/KeyboardEvent.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include "Event.h"
+
+#include <GLFW/glfw3.h>
+
+
+namespace Aquarius {
+
+    enum class KeyboardEvents
+    {
+        KeyPressedEvent, KeyRepeatedEvent, KeyReleasedEvent,
+    };
+
+    class KeyPressedEvent : public Event<KeyboardEvents>
+    {
+    public:
+        KeyPressedEvent() : Event<KeyboardEvents>(KeyboardEvents::KeyPressedEvent, m_Code, 0,0) {};
```

Comment:
Yeah I feel like this is very dangerous, Better to add the member variables in to these derived classes so that the callbacks can pass them in to these constructors when they create the appropriate event objects